includes/Modules/Analytics_4.php

```diff
@@ -1372,6 +1406,104 @@ function ( $order_def ) {
 		return $results;
 	}
 
+	/**
+	 * Parses the report response.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Data_Request                                   $data     Data request object.
+	 * @param Google_Service_AnalyticsData_RunReportResponse $response Request response.
+	 * @return mixed Parsed response data on success, or WP_Error on failure.
+	 */
+	protected function parse_report_response( Data_Request $data, $response ) {
+		// Return early if the response is not of the expected type.
+		if ( ! $response instanceof Google_Service_AnalyticsData_RunReportResponse ) {
+			return $response;
+		}
+
+		// Get report dimensions and return early if there is either more than one dimension or
+		// the only dimension is not "date".
+		$dimensions = $this->parse_reporting_dimensions( $data );
+		if ( count( $dimensions ) !== 1 || $dimensions[0]->getName() !== 'date' ) {
+			return $response;
+		}
+
+		// Get date ranges and return early if there is no date ranges for this report.
+		$date_ranges = $this->parse_reporting_dateranges( $data );
+		if ( empty( $date_ranges ) ) {
+			return $response;
+		}
+
+		$rows    = $response->getRows();
+		$metrics = $response->getMetricHeaders();
+
+		foreach ( $date_ranges as $date_range ) {
+			$start = strtotime( $date_range->getStartDate() );
+			$end   = strtotime( $date_range->getEndDate() );
+
+			// Skip this date range if either start date or end date is corrupted.
+			if ( ! $start || ! $end ) {
+				continue;
+			}
+
+			// Loop through all days in the date range and check if there is a metric value
+			// for it. If the metric value is missing, we will need to add one with a zero value.
+			$now = $start;
+			do {
+				$current_date = gmdate( 'Ymd', $now );
+
+				// Search for the current date in the response rows.
+				$found_dimenension = false;
+				foreach ( $rows as $row ) {
```

Comment:
Yes, i am not going to argue with you, you are right that associative arrays access is faster than iterating over all rows every time. Just wanted to say that in our case, the difference between implementations won't be visible even under a microscope because of a very small set of data that we have in our array. 😄

Reply:
Yup, agreed, it will be pretty minimal. Glad you've changed, but do agree it's not overly impactful, just seems like a slightly better choice to me.